tests/trainer/test_fsdp_checkpoint.py

```diff
@@ -629,11 +631,14 @@ def get_mono_state_dict_from_sharded_one(trainer):
     def compare_state_dicts():
         state_dict_from_trainer1 = mono_trainer.state.state_dict()
         state_dict_from_trainer2 = get_mono_state_dict_from_sharded_one(sharded_trainer)
-        _compare_model_params_between_state_dicts(state_dict_from_trainer1, state_dict_from_trainer2)
-        _compare_optims_between_state_dicts(state_dict_from_trainer1, state_dict_from_trainer2)
-        # Metrics are NOT equal as sharded checkpoints do not save or load metrics
-        # _compare_metrics_between_state_dicts(state_dict_from_trainer1, state_dict_from_trainer2)
-        _compare_timestamps_between_state_dicts(state_dict_from_trainer1, state_dict_from_trainer2)
+        # We are comparing full state dicts (all optim and model parameters are gathered on only rank 0)
+        # So we only need to compare on rank 0. Comparing on other ranks may cause errors because some state_dicts will be empty.
```

Comment:
```suggestion
        # We are comparing full state dicts (all optim and model parameters are gathered on only rank 0)
        # so we only need to compare on rank 0. Comparing on other ranks may cause errors because some state_dicts will be empty.
```